### PR TITLE
add basic formatter  support

### PR DESCRIPTION
**Pull request by assapir**
Part of #4.
Struggling a bit with how to add tests, as `console.log` will always return `undefine`. I guess that could be solved if we can pass a buffer to `printer` somehow? @MylesBorins

### CODE REVIEW

**Comment by MylesBorins** [comment]
hey @assapir I'm starting to sketch out how we could test console properly, it'll be a bit funky

**Comment by assapir** [comment]
I am  trying few things, the main one is to pass `io::Writer` in `Options`, which will by default stdout

**Comment by MylesBorins** [comment]
howdy howdy... is this something that we still want to pursue?

**Comment by assapir** [comment]
I didn't had anytime to look on it, sorry. I do still want to try to write tests somehow

**Comment by MylesBorins** [comment]
No prob and no rush. There are even more tests now you can use as examples

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "standard.enable": false
+}
```
Comment (MylesBorins) [inline .vscode/settings.json:3]: Is this a setting that can be set outside of the repo? I'd rather add .vscode to a .gitignore than add additional settings / configuration files to the project for a specific editor

File: core/src/lib.rs [hunk 2]
```diff
@@ -39,7 +39,7 @@ impl Options {
 }
 
 impl Default for Options {
-    fn default() -> Options {
+    fn default() -> Self {
```
Comment (MylesBorins) [inline core/src/lib.rs:42]: why did this change?

**Review by MylesBorins — APPROVED** [review]
LGTM

**Comment by MylesBorins** [comment]
For whatever reason this is not letting the actions run... I'm going to close this PR and open a new one and land it if CI is green.

**Comment by devsnek** [comment]
^ github actions have an outage rn https://githubstatus.com

**Pull request merged**